Assets/Scripts/Environmental/Room/RoomScript.cs

```diff
@@ -59,23 +59,37 @@ void OnEnemyDied(GameObject enemy)
      */
     void UnlockAllDoors()
     {
-        foreach (GameObject door in doors)
+        if(doors != null)
```

Comment:
Generally you want to [Fail-Fast](https://en.wikipedia.org/wiki/Fail-fast) in this sort of scenario, to avoid large nested if statements, by having a `if (doors == null) return;` at the start of this method. I won't force you to make these changes though, you can do them if you want.